src/main/java/seedu/multitasky/ui/uiutils/CommandAutocomplete.java

```diff
@@ -0,0 +1,103 @@
+package seedu.multitasky.ui.uiutils;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+
+import javafx.event.EventHandler;
+import javafx.fxml.FXML;
+import javafx.scene.control.TextField;
+import javafx.scene.input.KeyCode;
+import javafx.scene.input.KeyEvent;
+import javafx.scene.layout.Region;
+
+import seedu.multitasky.logic.commands.AddCommand;
+import seedu.multitasky.logic.commands.ClearCommand;
+import seedu.multitasky.logic.commands.DeleteCommand;
+import seedu.multitasky.logic.commands.EditCommand;
+import seedu.multitasky.logic.commands.ExitCommand;
+import seedu.multitasky.logic.commands.FindCommand;
+import seedu.multitasky.logic.commands.HelpCommand;
+import seedu.multitasky.logic.commands.HistoryCommand;
+import seedu.multitasky.logic.commands.ListCommand;
+import seedu.multitasky.logic.commands.SelectCommand;
+
+
+//@@author A0125586X
+/**
+ * Provides autocomplete functionality to the command box.
+ * If the string that the user has typed so far matches the beginning of only one command word,
+ * pressing the tab key will complete that command word automatically
+ */
+public class CommandAutocomplete {
+
+    private static final ArrayList<String> allCommandWords = new ArrayList<String>(Arrays.asList(new String[] {
+        AddCommand.COMMAND_WORD,
+        ClearCommand.COMMAND_WORD,
+        DeleteCommand.COMMAND_WORD,
+        EditCommand.COMMAND_WORD,
+        ExitCommand.COMMAND_WORD,
+        FindCommand.COMMAND_WORD,
+        HelpCommand.COMMAND_WORD,
+        HistoryCommand.COMMAND_WORD,
+        ListCommand.COMMAND_WORD,
+        SelectCommand.COMMAND_WORD
+    }));
+
+    @FXML
+    private Region commandBoxRegion;
+
+    @FXML
+    private TextField commandTextField;
+
+    private final EventHandler<KeyEvent> tabKeyEventHandler = new EventHandler<KeyEvent>() {
+        @Override
+        public void handle(KeyEvent event) {
+            if (event.getCode() == KeyCode.TAB) {
+                System.out.println("triggered");
```

Comment:
oops leftover from debugging haha. It's since been removed in my later commits, will be in my next PR